cpp/include/cudf/strings/split/split.hpp

```diff
@@ -78,5 +78,89 @@ std::unique_ptr<experimental::table> rsplit( strings_column_view const& strings,
                                              size_type maxsplit=-1,
                                              rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief The result(s) of a `contiguous_(r)split_record`
+ *
+ * Each column_view resulting from a split operation performed by
+ * contiguous_split_record will be returned wrapped in a
+ * `contiguous_split_record_result`. The column data addresses stored in the
+ * column_view objects are not owned by top level cudf::column objects. The
+ * backing memory is instead owned by the `all_data` field and in one contiguous
+ * block.
+ *
+ * The user is responsible for assuring that the `column_views` or any derived
+ * objects do not outlive the memory owned by `all_data`
+ */
+struct contiguous_split_record_result {
```

Comment:
Thank you for reviewing this,

and this struct mimics the existing _struct contiguous_split_result_.

https://github.com/rapidsai/cudf/blob/6351671f27c48753935dcee64f54e25b9885c52e/cpp/include/cudf/copying.hpp#L435

If we prefer to use std::pair instead of defining a new struct, I think we should modify both contiguous_split and contiguous_split_record to be consistent. I consider this is not something I can decide on my own; so I just decide to follow what already exists. Let me know your thoughts.

Reply:
No, it's fine. If this follows an already existent trend or an interface that was passed onto this task then please ignore and resolve.